src/main/java/me/magnet/consultant/Consultant.java

```diff
@@ -243,6 +246,13 @@ public Builder identifyAs(String serviceName, String datacenter, String hostname
 			return this;
 		}
 
+		public Builder withTag(String tag) {
+			if ( tag != null) {
```

Comment:
Shouldn't you just throw here, or maybe even let an NPE be triggered? If `tag == null` then that's most likely a programming error.